src/main/webapp/discussion.js

```diff
@@ -103,53 +129,77 @@ async function fetchDiscussion() {
 }
 
 /**
- * Creates an element for displaying {@code comment}.
+ * Renders a comment and its replies, with a form to post a new reply.
  */
-function createComment(comment) {
-  const element = document.createElement('li');
-  element.setAttribute(ATTR_ID, comment.commentKey.id);
+class DiscussionComment extends HTMLElement {
```

Comment:
This class confuses me. It changes all these variables, but hold no internal state.

Reply:
So this is a custom element template class, meaning it has a number of `slots` defined in the template.  It adds elements to itself, each specifying a `slot` to fill.  I attach the template as shadow-DOM which means this displays as the template, filling in all of the slots.  This class is purely for converting a `comment` into an element displayed on the page